Add Base64

CGenie opened the pull request:
Closes #152

mseri commented on src/lib/rpc.ml:
```diff
@@ -167,6 +169,8 @@ let rec to_string t =
       sprintf "{%s}"
         (map_strings ";" (fun (s, t) -> sprintf "%s:%s" s (to_string t)) ts)
+  | Base64 s ->
+      sprintf "S(%s)" s
```
Should let’s use `B64` Instead of `S` Here to distinguish the types

mseri commented on src/lib/jsonrpc.ml:
```diff
@@ -43,6 +43,7 @@ let rec rpc_to_json t =
   | Float r -> `Float r
   | String s -> `String s
   | DateTime d -> `String d
+  | Base64 b -> `String b
```
It’s really annoying that in json we don’t have a way to infer the type. Please add a comment on `json_to_rpc` as we have for the DateTime

mseri commented:
Thank you very much!

mseri commented on rpc.opam:
```diff
@@ -14,6 +14,7 @@ depends: [
   "rpclib-lwt" {=version}
   "ppx_deriving_rpc" {=version}
   "async" {>= "v0.9.0"}
+  "base64" {>= "3.4.0"}
```
This needs to be added to rpclib.opam

mseri replied:
rpc is a legacy virtual package

mseri commented:
I think also he ppxs need to be updated

CGenie commented:
@mseri Thank you for your comments, I updated the code. About ppxs, I'm not sure where to add this.
So far I just added this:
https://github.com/mirage/ocaml-rpc/pull/154/files#diff-36b922328d60097472c823ecbc21c056R15

mseri commented on ppx/common.ml:
```diff
@@ -11,7 +11,8 @@ let core_types loc =
     ; ("float", [%expr Rpc.Types.(Basic Float)])
-    ; ("bool", [%expr Rpc.Types.(Basic Bool)]) ]
+    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
+    ; ("base64", [%expr Rpc.Types.(Basic Base64)]) ]
```
Please revert this change. 
At this stage, base64 is not a core type. Like DateTime it needs to be handled by hand.

CGenie replied:
OK done.

mseri commented:
Just a small change and it's good to merge imo

mseri commented:
Thanks! I'll wait for Travis and merge

mseri commented on ppx/common.ml:
```diff
@@ -11,8 +11,7 @@ let core_types loc =
     ; ("float", [%expr Rpc.Types.(Basic Float)])
-    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
-    ; ("base64", [%expr Rpc.Types.(Basic Base64)]) ]
+    ; ("bool", [%expr Rpc.Types.(Basic Bool)]) ]) ]
```
```suggestion
    ; ("bool", [%expr Rpc.Types.(Basic Bool)])
    ]
```
